test/core.jl

```diff
@@ -5581,8 +5581,6 @@ let x5 = UnionField5(nothing, Int8(3))
     @test x5 == x5copy
     @test object_id(x5) === object_id(x5copy)
     @test hash(x5) === hash(x5copy)
-    @test pointer_from_objref(x5) === pointer_from_objref(x5)
-    @test pointer_from_objref(x5) !== pointer_from_objref(x5copy)
```

Comment:
The second test is always valid. Why isn't the first one working? It should be even easier for the interpreter than codegen.

Reply:
The first test does work, but I removed it since we had talked about disallowing `pointer_from_objref` on immutables. In the second test, `deepcopy` returns its argument for isbits objects, and the compiler re-boxed it but the interpreter didn't.